api/envoy/config/core/v3/health_check.proto

```diff
@@ -69,15 +69,24 @@ message HealthCheck {
     option (udpa.annotations.versioning).previous_message_type =
         "envoy.api.v2.core.HealthCheck.Payload";
 
+    enum DecoderType {
+      HEX = 0;
+      UTF8 = 1;
+    }
+
     oneof payload {
       option (validate.required) = true;
 
-      // Hex encoded payload. E.g., "000000FF".
-      string text = 1 [(validate.rules).string = {min_len: 1}];
+      // String payload. Default is Hex encoded payload, e.g., "000000FF".
```

Comment:
I don't think this field should have a default value, but I may be misunderstanding something.
IIUC there should either be `text` (and if so it should have a minimum length of 1) or `binary`.
If `text` is provided, there isn't a default value. If `binary` is provided, the `text` part shouldn't be accessed, so it doesn't need to have a default value.